src/themes/teams/components/Label/labelStyles.ts

```diff
@@ -1,42 +1,46 @@
-import { pxToRem } from '../../../../lib'
+import * as _ from 'lodash'
+
+import { pxToRem, generateColorScheme } from '../../../../lib'
 import { ComponentSlotStylesInput, ICSSInJSStyle } from '../../../types'
 import { LabelProps } from '../../../../components/Label/Label'
+import { LabelVariables } from './labelVariables'
 
-const labelStyles: ComponentSlotStylesInput<LabelProps, any> = {
-  root: ({ props: { image, imagePosition, circular }, variables }): ICSSInJSStyle => ({
-    padding: variables.padding,
-    ...(image &&
-      imagePosition === 'start' && {
-        paddingLeft: variables.startPaddingLeft,
-      }),
-    ...(image &&
-      imagePosition === 'end' && {
-        paddingRight: variables.endPaddingRight,
-      }),
-    display: 'inline-flex',
-    alignItems: 'center',
-    height: variables.height,
-    fontSize: pxToRem(14),
-    lineHeight: variables.height,
-    backgroundColor: variables.backgroundColor,
-    color: variables.color,
-    borderRadius: pxToRem(3),
-    ...(circular && {
-      borderRadius: variables.circularRadius,
-    }),
-    overflow: 'hidden',
-  }),
-  image: ({ variables }): ICSSInJSStyle => ({
-    height: variables.height,
-    width: variables.height,
-  }),
-  icon: ({ props }): ICSSInJSStyle => ({
-    ...(props.icon &&
-      typeof props.icon === 'object' &&
-      (props.icon as any).onClick && {
-        cursor: 'pointer',
+const labelStyles: ComponentSlotStylesInput<LabelProps, LabelVariables> = {
+  root: ({ props: p, variables: v }): ICSSInJSStyle => {
+    const colors = generateColorScheme(p.color, v.colorScheme)
```

Comment:
is it, essentially, `mergeColorSchemes`?

Reply:
also, this definitely shouldn't be a user responsibility to bother about this stuff. I would expect that `colors` object will be just directly consumed from the style function, like that (one of the options):

```jsx
root: ({ props, variables, colors }) => ({
   foreground: colors.foreground,
   ...
})
```

Reply:
I've added this with the latest commit (those are merged in renderComponent). Please take a look.